src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -604,44 +604,124 @@ internal BoundStatement BindLocalDeclarationStatement(LocalDeclarationStatementS
+                                                                             SyntaxNode node,
+                                                                             VariableDeclarationSyntax declarationSyntax,
+                                                                             out Conversion iDisposableConversion,
+                                                                             out MethodSymbol disposeMethod)
+        {
+            TypeSymbol iDisposable = this.Compilation.GetSpecialType(SpecialType.System_IDisposable);
+            iDisposableConversion = Conversion.NoConversion;
+            disposeMethod = null;
+            ImmutableArray<BoundLocalDeclaration> declarations;
 
-            if (variableCount == 1)
+            originalBinder.BindForOrUsingOrFixedDeclarations(declarationSyntax, LocalDeclarationKind.UsingVariable, diagnostics, out declarations);
+            Debug.Assert(!declarations.IsEmpty);
+
+            TypeSymbol declType = declarations[0].DeclaredType.Type;
+
+            if (declType.IsDynamic())
             {
-                return BindVariableDeclaration(kind, isVar, variableList[0], typeSyntax, declType, alias, diagnostics, node);
+                iDisposableConversion = Conversion.ImplicitDynamic;
             }
             else
             {
-                BoundLocalDeclaration[] boundDeclarations = new BoundLocalDeclaration[variableCount];
+                HashSet<DiagnosticInfo> useSiteDiagnostics = null;
+                iDisposableConversion = originalBinder.Conversions.ClassifyImplicitConversionFromType(declType, iDisposable, ref useSiteDiagnostics);
+                diagnostics.Add(declarationSyntax, useSiteDiagnostics);
 
-                int i = 0;
-                foreach (var variableDeclaratorSyntax in variableList)
+                if (!iDisposableConversion.IsImplicit)
                 {
-                    boundDeclarations[i++] = BindVariableDeclaration(kind, isVar, variableDeclaratorSyntax, typeSyntax, declType, alias, diagnostics);
+                    disposeMethod = TryFindDisposePatternMethod(declType, declarationSyntax, diagnostics);
+                    if (disposeMethod is null)
+                    {
+                        if (!declType.IsErrorType())
+                        {
+                            Error(diagnostics, ErrorCode.ERR_NoConvToIDisp, declarationSyntax, declType);
+                        }
+                        hasErrors = true;
+                    }
                 }
+            }
 
-                return new BoundMultipleLocalDeclarations(node, boundDeclarations.AsImmutableOrNull());
+            return new BoundUsingLocalDeclarations(node, disposeMethod, iDisposableConversion, declarations);
+        }
+
+        /// <summary>
+        /// Checks for a Dispose method on exprType in the case that there is no explicit
+        /// IDisposable conversion.
```

Comment:
Consider removing this part of the comment since this method does not check any conversion.